template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/RazorPagesWeb-CSharp/Pages/Account/Manage/Enable2fa.cshtml.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.Logging;
+using Company.WebApplication1.Data;
+
+namespace Company.WebApplication1.Pages.Account.Manage
+{
+    public class Enable2faModel : PageModel
+    {
+        private readonly UserManager<ApplicationUser> _userManager;
+        private readonly ILogger<Disable2faModel> _logger;
+
+        public Enable2faModel(UserManager<ApplicationUser> userManager,
+            ILogger<Disable2faModel> logger)
+        {
+            _userManager = userManager;
+            _logger = logger;
+        }
+
+       public async Task<IActionResult> OnGet()
+        {
+            var user = await _userManager.GetUserAsync(HttpContext.User);
+            if (user == null)
+            {
+                return RedirectToPage("/Error");
+            }
+
+            if (await _userManager.GetTwoFactorEnabledAsync(user))
+            {
+                return RedirectToPage("/Error");
+            }
+
+            if (!(await _userManager.GetValidTwoFactorProvidersAsync(user)).Any())
+            {
+                return RedirectToPage("/Error");
```

Comment:
The only way to get here in the already enabled state is to type the URL, so, meh, not really.